distributed/worker.py

```diff
@@ -3733,6 +3733,7 @@ def _ensure_computing(self) -> RecsInstrs:
     async def execute(self, key: str, *, stimulus_id: str) -> StateMachineEvent | None:
         if self.status in {Status.closing, Status.closed, Status.closing_gracefully}:
             return None
+        await self._started.wait()
```

Comment:
I'm also confused how this change fixes the plugin issue. From my read:
- `Worker._register_with_scheduler` connects to the scheduler and says "I'm here"
- In response, it [gets a list of plugins](https://github.com/dask/distributed/blob/7cd4c3478f4dfabb6e06a52f02a9ca3cba06d063/distributed/scheduler.py#L3672-L3677) to add
- The worker immediately [`await`s adding all those plugins](https://github.com/dask/distributed/blob/7cd4c3478f4dfabb6e06a52f02a9ca3cba06d063/distributed/worker.py#L1170-L1175)
- Only after the plugins have started, [`handle_scheduler` starts processing the messages from the scheduler](https://github.com/dask/distributed/blob/7cd4c3478f4dfabb6e06a52f02a9ca3cba06d063/distributed/worker.py#L1183)

`compute-task` messages are processed by `handle_scheduler` (they're not direct RPCs). So how can they be processed before the plugins have started? I don't see how there's actually a race condition here.

I wonder if what we're observing is actually that the worker plugin is getting sent to the scheduler after (some) workers have already started up. And tasks were already assigned to those workers that depended on the plugin. But that's not a dask bug, just user error.

Reply:
Yeah, 1000x repeats of the test you've added here are passing for me on `main`. I don't know if this is actually the issue.

Reply:
Hrm, that last comment is surprising.  This was failing for me before adding this change.  One moment.  Testing...

Reply:
Hrm, I'm seeing the same thing.  OK.  Closing.